packages/headless/src/features/triggers/triggers-slice.ts

```diff
@@ -17,5 +19,12 @@ export const triggerReducer = createReducer(
       state.redirectTo = redirectTriggers.length
         ? (redirectTriggers[0] as TriggerRedirect).content
         : '';
+
+      const queryTriggers: Trigger[] = action.payload.response.triggers.filter(
+        (trigger) => isTriggerQuery(trigger)
+      );
+      state.query = queryTriggers.length
+        ? (queryTriggers[0] as TriggerQuery).content
```

Comment:
I think if `isTriggerQuery` has the proper type-guard, you would not need this cast. You might also need to remove the `Trigger[]` type on `queryTriggers`, letting typescript implicitly type the variable for us,

Reply:
The cast is needed because the `Trigger` type can be any type of Trigger (Execute,  Notify, Query, Redirect). However the `content` property of all of these is a simple string except for `Execute` whose has `content: {name: string; params: {}[]};`

Reply:
On line 23 there is a filter step though. It should be possible to use either a more strict type for the `queryTriggers` variable, or no type at all, because we have filtered out the non-query triggers.

Reply:
On line 23, I can either put no type or specify the `Trigger` type. A more strict type results in an error since each `Trigger` has a different `type` value.

Reply:
```
export const triggerReducer = createReducer(
  getTriggerInitialState(),
  (builder) =>
    builder.addCase(executeSearch.fulfilled, (state, action) => {
      const redirectTriggers = action.payload.response.triggers.filter(
        isRedirectTrigger
      );
      state.redirectTo = redirectTriggers.length
        ? redirectTriggers[0].content
        : '';

      const queryTriggers = action.payload.response.triggers.filter(
        isQueryTrigger
      );
      state.query = queryTriggers.length ? queryTriggers[0].content : '';
    })
);
```

That should work!

Reply:
When trying this implementation, I get the following error for `state.redirectTo` and `state.query`: `Type '{ name: string; params: {}[]; }' is not assignable to type 'string'.` I still need to cast to the correct Trigger because `TriggerExecute` has `content: {name: string; params: {}[]}` instead of `content: string` like the rest.